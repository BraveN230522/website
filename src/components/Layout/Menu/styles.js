import styled from '@emotion/styled';

const MenuStyles = styled.ul`
  list-style-type: none;
  padding: 0;
  display: flex;

  ${({ theme }) => theme.media.tablet`
    display: none;
  `}
`;

const MenuItem = styled.li``;

const MenuLink = styled.a`
  border-bottom: 2px solid transparent;
  color: ${({ theme }) => theme.colors.white};
  cursor: pointer;
  display: block;
  font-family: ${({ theme }) => theme.fonts.DINAlternate};
  font-size: 21px;
  font-weight: 400;
  letter-spacing: -0.14px;
  line-height: 28px;
  margin: 0 16px;
  text-align: center;
  transition: all 0.3s ease-in;

  &:hover {
    text-decoration: none;
    border-bottom: 2px solid ${({ activeLink, theme }) => (activeLink ? theme.colors.swiamBlue : theme.colors.white)};
  }

  & {
    opacity: ${({ activeLink }) => activeLink && 1};
    ${({ theme, activeLink }) =>
      activeLink &&
      `
        border-bottom: 2px solid ${theme.colors.swiamBlue};
      `};
  }
`;

export { MenuStyles, MenuItem, MenuLink };
