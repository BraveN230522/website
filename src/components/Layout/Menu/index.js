import React from 'react';
import { useRouter } from 'next/router';
import Link from 'next/link';

import { MenuStyles, MenuItem, MenuLink } from './styles';

const mockMenu = [
  { id: 1, menu: 'Events', href: '/events' },
  { id: 2, menu: 'Cities', href: '/city/21' },
  { id: 3, menu: 'Hotels', href: '/hotels' },
  // { id: 4, menu: 'Experiences', href: '/experience/1' },
  { id: 4, menu: 'About', href: '/about' },
];

const PAGE_NAME_REGEX = /^(\/\w+)/;

const Menu = () => {
  const router = useRouter();

  const getActiveLink = url => {
    const urlMatch = url.match(PAGE_NAME_REGEX);
    const pathNameMatch = router.pathname.match(PAGE_NAME_REGEX);

    return urlMatch && pathNameMatch && urlMatch[0] === pathNameMatch[0];
  };

  return (
    <MenuStyles>
      {mockMenu.map(item => (
        <MenuItem key={item.id}>
          <Link href={item.href} passHref>
            <MenuLink activeLink={getActiveLink(item.href)}>{item.menu}</MenuLink>
          </Link>
        </MenuItem>
      ))}
    </MenuStyles>
  );
};

export default Menu;
