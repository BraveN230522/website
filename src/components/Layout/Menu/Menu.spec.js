import React from 'react';
import { render } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import Menu from './index';

describe('<Menu />', () => {
  it('Renders a Menu component', () => {
    const { container } = render(<Menu />, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });
});
