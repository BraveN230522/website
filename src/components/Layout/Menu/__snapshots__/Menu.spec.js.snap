// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Renders a Menu component 1`] = `
.emotion-20 {
  list-style-type: none;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:60em) {
  .emotion-20 {
    display: none;
  }
}

.emotion-0 {
  border-bottom: 2px solid transparent;
  color: #ffffff;
  cursor: pointer;
  display: block;
  font-family: DINAlternateRegular,sans-serif;
  font-size: 21px;
  font-weight: 400;
  -webkit-letter-spacing: -0.14px;
  -moz-letter-spacing: -0.14px;
  -ms-letter-spacing: -0.14px;
  letter-spacing: -0.14px;
  line-height: 28px;
  margin: 0 16px;
  text-align: center;
  -webkit-transition: all 0.3s ease-in;
  transition: all 0.3s ease-in;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 2px solid #ffffff;
}

<div>
  <ul
    class="emotion-20 emotion-21"
  >
    <li
      class="emotion-2 emotion-3"
    >
      <a
        class="emotion-0 emotion-1"
        href="/events"
      >
        Events
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        class="emotion-0 emotion-1"
        href="/city/21"
      >
        Cities
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        class="emotion-0 emotion-1"
        href="/hotels"
      >
        Hotels
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        class="emotion-0 emotion-1"
        href="/experience/1"
      >
        Experiences
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        class="emotion-0 emotion-1"
        href="/about"
      >
        About
      </a>
    </li>
  </ul>
</div>
`;
